pep-9999.rst

```diff
@@ -0,0 +1,501 @@
+PEP: 9999
+Title: tomllib: Support for parsing TOML in the Standard Library
+Author: Taneli Hukkinen, Shantanu Jain <hauntsaninja at gmail.com>
+Sponsor: Petr Viktorin <encukou@gmail.com>
+Discussions-To: https://discuss.python.org/t/adopting-recommending-a-toml-parser/4068
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 01-Jan-2022
+Python-Version: 3.11
+Post-History: 1900-01-01
+
+
+Abstract
+========
+
+This proposes adding a module, ``tomllib``, to the standard library for
+parsing TOML (Tom's Obvious Minimal Language,
+`https://toml.io <https://toml.io/en/>`_).
+
+
+Motivation
+==========
+
+The TOML format is the format of choice for Python packaging, as evidenced by
+:pep:`517`, :pep:`518` and :pep:`621`. Including TOML support in the standard
+library helps avoid bootstrapping problems for Python build tools. Currently
+most Python build tools need to vendor a TOML parsing library.
+
+Python tools are increasingly configurable via TOML, for examples: ``black``,
+``mypy``, ``pytest``, ``tox``, ``pylint``, ``isort``. Those that are not, such
+as ``flake8``, cite the lack of standard library support as a `main reason why
+<https://github.com/PyCQA/flake8/issues/234#issuecomment-812800657>`_.
+
+Given the special place TOML already has in the Python ecosystem, it makes sense
+for this to be an included battery.
```

Comment:
I would suggest getting buy-in from the maintainers of tools that support `pyproject.toml`, e.g. poetry, pdm, and the tools mentioned above.

Reply:
`mypy` should be easy seeing that @hauntsaninja is a maintainer :smile: 

`flit` is not currently listed but I'm fairly sure I won't have to persuade @takluyver ?

`isort` has a no-dependencies policy (vendor instead) so I bet @timothycrosley also has no objections?

Reply:
As a maintainer of Black and mypy I'd also be happy to see a TOML parser in the stdlib :)

Reply:
I expect @pradyunsg could do so for pip and some other packaging tools?

pip switched vendoring toml to tomli: https://github.com/pypa/pip/pull/10035

Reply:
I'd be in favour as a pip maintainer.